File: travel_cost_calculator.py
# Name: Chnaduka P.A.K      Index : EG/2020/3865


from csv import reader

# Initialize dictionaries to store data from CSV files
hotel_rates = {}
exchange_rates = {}
flight_costs = {}

# Function to load hotel rates from a CSV file
def load_hotel_rates(file_path):
    with open(file_path) as hotel_file:
        csv_reader = reader(hotel_file)
        for row in csv_reader:
            hotel_rates[row[0]] = float(row[1])

# Function to load exchange rates from a CSV file
def load_exchange_rates(file_path):
    with open(file_path) as exchange_file:
        csv_reader = reader(exchange_file)
        for row in csv_reader:
            exchange_rates[row[0].upper()] = float(row[1])

# Function to load flight costs from a CSV file
def load_flight_costs(file_path):
    with open(file_path) as flight_file:
        csv_reader = reader(flight_file)
        for row in csv_reader:
            flight_costs[row[0]] = float(row[1])

# Main function for the program
def main():
    # Load data from CSV files
    load_hotel_rates('data/hotel_rates.csv')
    load_exchange_rates('data/exchange_rates.csv')
    load_flight_costs('data/flight_costs.csv')

    # Get user input for the destination
    destination = input("Enter your destination: ").upper()

    # Calculate flight and hotel costs
    flight_cost = flight_costs.get(destination, 0.0)
    hotel_cost = hotel_rates.get(destination, 0.0)

    # Get user input for the stay duration
    days = int(input("Enter your stay duration in days: "))
    
    # Calculate the total hotel cost for the specified duration
    total_hotel_cost = hotel_cost * days
    
    # Calculate the total cost (flight + hotel)
    total_cost = flight_cost + total_hotel_cost

    # Display the costs in USD
    print(f"Flight cost: USD {flight_cost:.2f}")
    print(f"Hotel cost for {days} days: USD {total_hotel_cost:.2f}")
    print(f"Total: USD {total_cost:.2f}")

    # Get user input for the desired currency
    currency = input(f"Select your currency for final price estimation ({', '.join(exchange_rates.keys())}): ")

    # Convert the total cost to the selected currency
    price_in_selected_currency = total_cost * exchange_rates[currency]
    print(f"Total in {currency}: {price_in_selected_currency:.2f}")

if __name__ == "__main__":
    main()

File: TravelCostCalculator.java
// ABEYSEKARA P.K.
// EG/2020/3799

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;

public class TravelCostCalculator {
    // Initialize maps to store hotel rates, exchange rates, and flight costs
    static Map<String, Double> hotelRates = new HashMap<>();
    static Map<String, Double> exchangeRates = new HashMap<>();
    static Map<String, Double> flightCosts = new HashMap<>();

    // Method to read hotel rates from a file and populate the hotelRates map
    static void readHotelRates(String file) throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader(file));
        String line;
        while ((line = reader.readLine()) != null) {
            String[] parts = line.split(",");
            hotelRates.put(parts[0].toUpperCase(), Double.parseDouble(parts[1]));
        }
        reader.close();
    }

    // Method to read exchange rates from a file and populate the exchangeRates map
    static void readExchangeRates(String file) throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader(file));
        String line;
        while ((line = reader.readLine()) != null) {
            String[] parts = line.split(",");
            exchangeRates.put(parts[0].toUpperCase(), Double.parseDouble(parts[1]));
        }
        reader.close();
    }

    // Method to read flight costs from a file and populate the flightCosts map
    static void readFlightCosts(String file) throws IOException {
        BufferedReader reader = new BufferedReader(new FileReader(file));
        String line;
        while ((line = reader.readLine()) != null) {
            String[] parts = line.split(",");
            flightCosts.put(parts[0].toUpperCase(), Double.parseDouble(parts[1]));
        }
        reader.close();
    }

    public static void main(String[] args) {
        try {
            // Read data files and populate maps
            readHotelRates("data/hotel_rates.csv");
            readExchangeRates("data/exchange_rates.csv");
            readFlightCosts("data/flight_costs.csv");

            // Get user input for destination and stay duration
            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
            System.out.print("Enter your destination: ");
            String destination = reader.readLine().toUpperCase();

            System.out.print("Enter your stay duration in days: ");
            int stayDuration = Integer.parseInt(reader.readLine());

            // Calculate total cost in USD
            double flightCost = flightCosts.getOrDefault(destination, 0.0);
            double hotelCost = hotelRates.getOrDefault(destination, 0.0) * stayDuration;
            double totalCostUSD = flightCost + hotelCost;

            // Display costs in USD
            System.out.printf("Flight cost: USD %.2f\n", flightCost);
            System.out.printf("Hotel cost (%d days): USD %.2f\n", stayDuration, hotelCost);
            System.out.printf("Total: USD %.2f\n", totalCostUSD);

            // Get user input for currency and calculate final price in local currency
            String[] availableCurrencies = exchangeRates.keySet().toArray(new String[0]);
            System.out.print("Select your currency for final price estimation(" + String.join(", ", availableCurrencies) + "): ");
            String selectedCurrency = reader.readLine().toUpperCase();
            double finalPriceLocalCurrency = totalCostUSD * exchangeRates.get(selectedCurrency);

            // Display final price in local currency
            System.out.printf("Total in %s: %.2f\n", selectedCurrency, finalPriceLocalCurrency);

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
